### PR TITLE
[Java] Fixed build by adding more checks for int8, int16

### PR DESCRIPTION
**Pull request by razajafri**
Add additional checks for int8, int16

fixes [#/rapidsai/cudf/4127](https://github.com/NVIDIA/spark-rapids/issues/4127)

### CODE REVIEW

**Comment by razajafri** [comment]
rerun tests

**Review by revans2 — APPROVED** [review]

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 1]
```diff
@@ -398,12 +398,13 @@ public final ColumnVector isInteger() {
    * for null entries.
    *
    * @param intType the data type that should be used for bounds checking. Note that only
-   *                integer types are allowed.
+   *                integer types are allowed including (int8, int16, int32, int64).
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:401]: This is not correct and that type list may not be needed. You missed unsigned types.

Comment (razajafri) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:401]: From a Java developers point of view integer types will not include bytes, shorts or longs. I am trying to highlight to the dev that this isn't just Java int its a cpp/cudf int. which does include signed/unsigned int8 - int64. 
How does this sound `Note that only cudf integer types are allowed including signed/unsigned int8 through int64`

Comment (ttnghia) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:401]: Yes that is great.

**Review by ttnghia — APPROVED** [review]

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by razajafri** [comment]
@gpucibot merge

**Pull request merged**